diesel_cli/src/migrations/mod.rs

```diff
@@ -273,55 +273,75 @@ fn redo_migrations<Conn, DB>(
+                harness.revert_all_migrations(migrations_dir.clone())?
+            } else {
+                let number = args.get_one::<u64>("REDO_NUMBER").unwrap();
+
+                (0..*number)
+                    .filter_map(|_| {
+                        match harness.revert_last_migration(migrations_dir.clone()) {
+                            Ok(v) => Some(Ok(v)),
+                            Err(e) if e.is::<MigrationError>() => {
+                                match e.downcast_ref::<MigrationError>() {
+                                    // If n is larger then the actual number of migrations,
+                                    // just stop reverting them
+                                    Some(MigrationError::NoMigrationRun) => None,
+                                    _ => Some(Err(e)),
+                                }
                             }
+                            Err(e) => Some(Err(e)),
                         }
-                        Err(e) => {
-                            Some(Err(e))
-                        }
-                    }
-                })
-                .collect::<Result<Vec<_>, _>>()?
-        };
+                    })
+                    .collect::<Result<Vec<_>, _>>()?
+            };
 
-        let mut migrations = MigrationSource::<DB>::migrations(&migrations_dir)?
-            .into_iter()
-            .map(|m| (m.name().version().as_owned(), m))
-            .collect::<HashMap<_, _>>();
+             let mut migrations = MigrationSource::<DB>::migrations(&migrations_dir)
+                 .unwrap_or_else(handle_error)
+                 .into_iter()
+                 .map(|m| (m.name().version().as_owned(), m))
+                 .collect::<HashMap<_, _>>();
 
-        let mut migrations = reverted_versions
-            .into_iter()
-            .map(|v| {
-                migrations
-                    .remove(&v)
-                    .ok_or_else(|| MigrationError::UnknownMigrationVersion(v.as_owned()))
-            })
-            .collect::<Result<Vec<_>, _>>()?;
-        // Sort the migrations by version to apply them in order.
-        migrations.sort_by_key(|m| m.name().version().as_owned());
+            let mut migrations = reverted_versions
+                .into_iter()
+                .map(|v| {
+                    migrations
+                        .remove(&v)
+                        .ok_or_else(|| MigrationError::UnknownMigrationVersion(v.as_owned()))
+                })
+                .collect::<Result<Vec<_>, _>>()?;
+            // Sort the migrations by version to apply them in order.
+            migrations.sort_by_key(|m| m.name().version().as_owned());
 
-        harness.run_migrations(&migrations)?;
+            harness.run_migrations(&migrations)?;
 
-        Ok(())
-    };
+            Ok(())
+        };
 
-    if should_redo_migration_in_transaction(conn) {
+    if !should_use_not_use_transaction && should_redo_migration_in_transaction(conn) {
```

Comment:
since there are circumstances now when migrations won't be redone as a transaction, the comment on line 266 and 267 needs to be changed.

Reply:
Will correct that